src/benchInsert.c

```diff
@@ -520,6 +520,80 @@ int32_t getVgroupsOfDb(SBenchConn *conn, SDataBase *database) {
     return vgroups;
 }
 
+#ifdef WEBSOCKET
+int32_t getVgroupsWS(SBenchConn *conn, SDataBase *database) {
+    int vgroups = 0;
+    char sql[128] = "\0";
+    snprintf(sql, sizeof(sql),
+             g_arguments->escape_character
+                 ? "SHOW `%s`.VGROUPS"
+                 : "SHOW %s.VGROUPS",
+             database->dbName);
+
+    // query
+    WS_RES *res = ws_query_timeout(conn->taos_ws, sql, g_arguments->timeout);
+    int32_t code = ws_errno(res);
+    if (code != 0) {
+        // failed
+        errorPrint("Failed ws_query_timeout <%s>, code: 0x%08x, reason: %s\n",
+                   sql, code, ws_errstr(res));
+        ws_free_result(res);           
+        return 0;
+    }
+
+    // fetch
+    WS_ROW row;
+    database->vgArray = benchArrayInit(8, sizeof(SVGroup));
+    while ( (row = ws_fetch_row(res)) && !g_arguments->terminate) {
+        SVGroup *vg = benchCalloc(1, sizeof(SVGroup), true);
+        vg->vgId = *(int32_t *)row[0];
+        benchArrayPush(database->vgArray, vg);
+        vgroups++;
+        debugPrint(" ws fetch vgroups vgid=%d cnt=%d \n", vg->vgId, vgroups);
+    }
+    ws_free_result(res);
+    database->vgroups = vgroups;
+
+    // return count
+    return vgroups;
+}
+
+
+int32_t getTableVgidWS(SBenchConn *conn, char *db, char *tb, int32_t *vgId) {
+    char sql[128] = "\0";
+    snprintf(sql, sizeof(sql),
+                 "select vgroup_id from information_schema.ins_tables where db_name='%s' and table_name='%s';",
+                 db, tb);
+    // query
+    WS_RES *res = ws_query_timeout(conn->taos_ws, sql, g_arguments->timeout);
+    int32_t code = ws_errno(res);
+    if (code != 0) {
+        // failed
+        errorPrint("Failed ws_query_timeout <%s>, code: 0x%08x, reason: %s\n",
+                   sql, code, ws_errstr(res));
+        ws_free_result(res);           
+        return code;
+    }
+
+    // fetch
+    WS_ROW row;
+    while ( (row = ws_fetch_row(res)) && !g_arguments->terminate) {
+        *vgId = *(int32_t *)row[0];
+        debugPrint(" getTableVgidWS table:%s vgid=%d\n", tb, *vgId);
+        break;
+    }
+    ws_free_result(res);
+
+    if(*vgId == 0) {

```

Comment:
The pointer vgId points to something that does not show initialization, either inside this function or at the call level

Reply:
already initialize variant vgId = 0, fixed